File: berceau/berceau-python/Dispositif.py

class Dispositif:
    def __init__(self,pin,name,etat):
        self.pin=pin
        self.name=name
        self.etat=etat

    
    def initialiser(self,pin):
        pass

    def afficherInfo(self):
        if(self.etat):
            etat="marche"
        else:
            etat="arrete"

        return (f"pin utiliser est : {self.pin} et name de dispositif est {self.name} et etat {etat}")
        

File: mobile/android/app/src/main/java/com/example/appmobile/model/entity/Dispositif.java
package com.example.appmobile.model.entity;


import java.io.Serializable;

public  class  Dispositif implements Serializable {

    private int pin;
    private String name;
    private Boolean etat;


    public Dispositif() {
    }

    public Dispositif(int pin, String name, Boolean etat) {
        this.pin = pin;
        this.name = name;
        this.etat = etat;
    }

    public int getPin() {
        return pin;
    }

    public void setPin(int pin) {
        this.pin = pin;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public Boolean getEtat() {
        return etat;
    }

    public void setEtat(Boolean etat) {
        this.etat = etat;
    }

    public void initialiser(int pin, String name) {
        System.out.println("la dispositif "+name+" est donctionnement dans pin "+pin);
        this.etat = true;
    }



    public String afficherInfo() {
        return "pin : "+pin+" name : "+name + " etat : "+etat;
    }



}
